src/js/requestHandler.js

```diff
@@ -0,0 +1,33 @@
+const requestHandler = {
+    
+    initRequest(request, component) {
+
+        request.addEventListener("load", () => {
+            return component.setState({
+                ajaxResponse: JSON.parse(request.response),
+                errorOccured: false
+            });
+        });
+
+        request.addEventListener("error", () => {
+            return component.setState({
+                ajaxResponse: [],
+                errorOccured: true
+            });
+        });
+
+        request.addEventListener("readystatechange", () => {
+            if(request.readyState <= 1) {
```

Comment:
use the actual values instead of numerical values

Reply:
readyState return numbers, and request doesn't have a property with a string version of the values